### ISSUE TITLE
nixos/bookstack: all secrets are replaced by app_key

### ISSUE THREAD

**Issue by ymarkus** [NixOS/nixpkgs#155605]
### Describe the bug
in #154564 secret management was redone for bookstack. Now I can't place a mail password as secret. It's also appended with `Mail`:
```console
❯ cat .env
APP_CONFIG_CACHE=/run/bookstack/cache/config.php
APP_EVENTS_CACHE=/run/bookstack/cache/events.php
APP_KEY=<<APPKEY>>
APP_PACKAGES_CACHE=/run/bookstack/cache/packages.php
APP_ROUTES_CACHE=/run/bookstack/cache/routes-v7.php
APP_SERVICES_CACHE=/run/bookstack/cache/services.php
APP_URL=https://wiki.example.com
DB_DATABASE=bookstack
DB_HOST=localhost
DB_PORT=3306
DB_USERNAME=bookstack
MAIL_DRIVER=smtp
MAIL_FROM=wiki@example.com
MAIL_FROM_NAME=Wiki
MAIL_HOST=mail.example.com
MAIL_PASSWORD=<<APPKEY>>Mail
MAIL_PORT=587
MAIL_USERNAME=wiki@example.com
SESSION_SECURE_COOKIE=true
```

### Notify maintainers
@talyz

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.91, NixOS, 21.11 (Porcupine)`
```

**Comment by talyz** [comment]
Oh, sorry about that. I can't see any obvious culprit in the code and I really don't know where `Mail` came from. Can you paste the service script as it's been rendered? Can you paste a configuration that reproduces this?

**Comment by talyz** [comment]
Oh, I suppose this could happen if the mail password file path is a superset of the app key file path. Is that the case?

**Comment by talyz** [comment]
I've submitted a fix for you to test: #157719. I hope it solves this issue.
